src/nrnpython/inithoc.cpp

```diff
@@ -207,6 +207,33 @@ static int have_opt(const char* arg) {
     return 0;
 }
 
+#if __linux__ || DARWIN
+
+/* we do this because thread sanitizer does not allow system calls.
+   In particular
+      system("stty sane")
+   returns an error code of 139
+*/
+
+#include <iostream>
+#include <termios.h>
+#include <unistd.h>
+
+static struct termios original_termios;
+
+static void save_original_terminal_settings() {
+    if (tcgetattr(STDIN_FILENO, &original_termios) == -1) {
+        std::cerr << "Error getting original terminal attributes\n";
```

Comment:
Your'e right. I'll change it.

Reply:
Hmm. This is trickier than I thought. ```utils/logger.hpp``` requires ```fmt/printf.h``` and I don't know if that is a good idea for this particular python module file. And I don't know if we are allowed to link to ```nrnpy_pr_stdoe_callback``` if we are dynamically loading.

Reply:
After some searching, it is still not clear to me the circumstances for tcgetattr/tsetattr returning a non-zero error code. It may have something to do with whether STDIN_FILENO was closed by the user. I'm leaning toward leaving in the ```std::cerr``` messages. At least there won't be any warnings about ignoring return values.
Consider this observation:
```
$ python -i -c 'from neuron import h, gui' < /dev/null
Error getting original terminal attributes
>>> 
$
```
I added to the check ``` && isatty(STDIN_FILENO)```